src/providers/FileSystemProvider/TextSearchProvider.ts

```diff
@@ -104,13 +104,46 @@ export class TextSearchProvider implements vscode.TextSearchProvider {
     } else {
       const sysStr = params.has("system") && params.get("system").length ? params.get("system") : "0";
       const genStr = params.has("generated") && params.get("generated").length ? params.get("generated") : "0";
+
+      let uri = options.folder;
+      const uriQuery = new URLSearchParams(uri.query);
+
+      if (!uriQuery.has("filter")) {
+        // Unless isfs spec already includes a filter (which it rarely does), apply includes and excludes at the server side.
+        // If include or exclude field is set to, say, A1B2M*.int there will be two consecutive options.[in|ex]cludes elements:
+        //   **/A1B2M*.int/**
+        //   **/A1B2M*.int
+        //
+        // Ignore first, and strip **/ prefix from second.
+        // When 'Use Exclude Settings and Ignore Files' is enabled (which is typical) options.excludes will also contain entries from files.exclude and search.exclude settings.
+        // This will result in additional server-side filtering which is probably superfluous but harmless (other than perhaps incurring a performance cost, probably small).
+        const tidyFilters = (filters: string[]): string[] => {
+          return filters
+            .map((value, index, array) =>
+              value.endsWith("/**") && index < array.length - 1 && array[index + 1] + "/**" === value
+                ? ""
+                : value.startsWith("**/")
+                ? value.slice(3)
+                : value
+            )
+            .filter((value) => value !== "");
+        };
+        const tidiedIncludes = tidyFilters(options.includes);
+        const tidiedExcludes = tidyFilters(options.excludes);
+        const filter = tidiedIncludes.join(",") + (tidiedExcludes.length === 0 ? "" : ",'" + tidiedExcludes.join(",'"));
+        if (filter) {
+          uriQuery.append("filter", filter);
+          uri = options.folder.with({ query: uriQuery.toString() });
```

Comment:
This actually could lead to more searching, specifically if the user has a type set that doesn't match what they're including. I.e. `?type=cls` and includes contains `*.mac`.

Reply:
I don't think it's practical to try and respect a `?type=xxx` qualifier as well as a "files to include" spec in Search. User needs to be explicit about file extension if they want search results to only be a particular type.

I have pushed a fix for https://github.com/intersystems-community/vscode-objectscript/issues/1019#issuecomment-1333661047